File: SantaFinder.Web/Models/ChangeOrder/OrderPostInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using SantaFinder.Entities;
using SantaFinder.Web.Models.OrderHistory;
using SantaFinder.Web.Models.Shared;

namespace SantaFinder.Web.Models.ChangeOrder
{
    public class OrderPostInfo
    {
        public string ChildrenNames { get; set; }
        public DateTime Datetime { get; set; }
        public AddressWithLocation Address { get; set; }
        public IEnumerable<PresentInfoForEdit> Presents { get; set; }
    }
}

File: SantaFinder.Web/Client/src/app/client/order-info/order-post-info.ts
import { Address, Location } from '../../core/models';
import { PresentInfoForEditing } from './present-info-for-editing';

export interface OrderPostInfo {
    childrenNames: string;
    datetime: Date;
    address: {
        line: Address;
        location: Location;
    };
    presents: PresentInfoForEditing[];
}
